embark/users/views.py

```diff
@@ -90,3 +99,71 @@ def signout(request):
+def password_change(request):
+    if request.method == "POST":
+        logger.debug(request.POST)
+        user = get_user(request)
+
+        data = {k: v[0] for k, v in dict(request.POST).items()}
+        logger.debug(data)
+        try:
+            body = {k: v[0] for k, v in dict(request.POST).items()}
+            try:
+                oldPassword = body['oldPassword']
+                newPassword = body['newPassword']
+                confirmPassword = body['confirmPassword']
+
+                if user.check_password(oldPassword):
+                    if oldPassword == newPassword:
+                        logger.debug('New password = old password')
+                        return render(request, 'uploader/passwordChange.html',
+                                      {'error_message': True, 'message': 'New password matches the old password'})
+                    if newPassword == confirmPassword:
+                        user.set_password(newPassword)
+                        user.save()
+                        authenticate(request, username=user.username, password=newPassword)
+                        login(request, user)
+                        logger.debug('New password set, user authenticated')
+                        return render(request, 'uploader/passwordChangeDone.html',
+                                      {'success_message': True, 'message': 'Password change successful.'})
+                    else:
+                        logger.debug('Passwords do not match')
+                        return render(request, 'uploader/passwordChange.html',
+                                      {'error_message': True, 'message': 'Passwords do not match.'})
+                else:
+                    logger.debug('Old password is incorrect')
+                    return render(request, 'uploader/passwordChange.html',
+                                  {'error_message': True, 'message': 'Old password is incorrect.'})
+            except KeyError:
+                logger.exception('Missing keys from data-passwords')
+                return render(request, 'uploader/passwordChange.html',
+                              {'error_message': True, 'message': 'Some fields are empty!'})
+        except Exception as error:
+            logger.exception('Wide exception in Password Change: %s', error)
+            return render(request, 'uploader/passwordChange.html', {'error_message': True,
+                                                                    'message': 'Something went wrong when changing the password for the user.'})
+    else:
+        return render(request, 'uploader/passwordChange.html')
+
+
+@csrf_exempt
+@login_required(login_url='/' + settings.LOGIN_URL)
+@require_http_methods(["GET", "POST"])
+def acc_delete(request):
+    if request.method == "POST":
+        logger.debug('disabling account')
+        user = get_user(request)
+        logger.debug('' + datetime.now().strftime("%H:%M:%S"))
```

Comment:
users/views.py:161:8: W1201: Use lazy % formatting in logging functions (logging-not-lazy)

-> there should be a way to bring this into the form of:
     `logger.debug(' %s Account: %s disabled', time, user )`